rclpy/rclpy/context.py

```diff
@@ -149,3 +152,18 @@ def get_domain_id(self):
             raise RuntimeError('Context must be initialized before it can have a domain id')
         with self.__context, self._lock:
             return self.__context.get_domain_id()
+
+    def __enter__(self) -> "Context":
+        # We do not accept parameters here. If one wants to customize the init() call,
+        # they would have to call it manaully and not use the ContextManager convenience
+        self.init()
```

Comment:
Ah, this makes me wish we had put the arguments to `init()` in `__init__` instead. Then users could have done:

```python3

with Context(initialize_logging=false) as ctx:
   ...
```

Not asking for you to fix this in this Pr, but I'm interested if you have any ideas.

Reply:
Yeah, and for me personally, the many `init()` calls are quite unexpected as a whole. As a Python user, I usually expect the `__init__()` to actually give me a fully initialized instance. I guess that's kind of like Resource acquisition is initialization (RAII) that I expect.

I guess you could gracefully transition to the `__init__()` by having the following interim solution for one release or something:
- Have `__init__()` take the arguments that `init()` takes, and if they are actually passed set things up right away. If they are not passed (which might need a trick to be detected), the user would need to call `init()`.
- The `init()` complains if the `__init__()` already set things up. If that was not the case, set things up here but show a deprecation warning pointing the user to move to `__init__()`.

Overall, I'm not sure whether it would even impact too many non-core ROS2 packages out there. I suppose there is often no need to handle `Context` object directly.

Maybe it makes sense to move this discussion into its own issue.